sdk/mobile-center-distribute/src/main/java/com/microsoft/azure/mobile/distribute/Distribute.java

```diff
@@ -1130,6 +1173,51 @@ public void onClick(DialogInterface dialog, int which) {
     }
 
     /**
+     * Show update setupp failed dialog.
+     */
+    @UiThread
+    private synchronized void showUpdateSetupFailedDialog(final String errorMessage) {
+        /* Check if we need to replace dialog. */
+        if (!shouldRefreshDialog(mUpdateSetupFailedDialog)){
+            return;
+        }
+        MobileCenterLog.debug(LOG_TAG, "Show update setup failed dialog.");
+
+        if (mForegroundActivity != null) {
+            AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(mForegroundActivity);
+            dialogBuilder.setCancelable(false);
+            dialogBuilder.setTitle(R.string.mobile_center_distribute_update_failed_dialog_title);
+            dialogBuilder.setMessage(errorMessage);
+            dialogBuilder.setPositiveButton(R.string.mobile_center_distribute_update_failed_dialog_ignore, new DialogInterface.OnClickListener() {
+
+                @Override
+                public void onClick(DialogInterface dialog, int which) {
+                    PreferencesStorage.putString(PREFERENCE_KEY_UPDATE_SETUP_FAILED_PACKAGE_HASH_KEY, DistributeUtils.computeReleaseHash(mPackageInfo));
+                }
+            });
+            dialogBuilder.setNegativeButton(R.string.mobile_center_distribute_update_failed_dialog_reinstall, new DialogInterface.OnClickListener() {
+
+                @Override
+                public void onClick(DialogInterface dialog, int which) {
+                    String url = mInstallUrl;
+                    // Add a flag to the install url to indicate that the update setup failed, to show a help page
```

Comment:
Add empty line before comment.

Reply:
and comment is always preceding any code block, you reversed the 2 lines here.